src/java_bytecode/java_string_library_preprocess.h

```diff
@@ -52,7 +52,9 @@ class java_string_library_preprocesst:public messaget
   void add_string_type(const irep_idt &class_name, symbol_tablet &symbol_table);
   bool is_known_string_type(irep_idt class_name);
 
+#ifndef UNIT_TEST
```

Comment:
I don't think this is good practice. What does @peterschrammel think about it? It looks OK now, but we might end up with a mess of `#ifdef` if we go down that path. Is there any way around that?

Reply:
This is relatively harmless way to do it without doing heavy duty refactoring. Making it private would compromise the interface. Inheriting would disallow specifying class as final. Friend can't be used, because Catch doesn't specify method names.

The fact that this macro is used signals the need for splitting the class into more manageable parts, but I don't think it has to be done immediately.

Reply:
I chose the solution of declaring a friend function inside a `#ifdef CBMC_UNIT_TEST` and change the Makefile of `unit/` to define `CBMC_UNIT_TEST`